R/session_check_reauth.R

```diff
@@ -0,0 +1,61 @@
+#'@title Checks current session and re-authenticates if necessary
+#'
+#'@inheritParams authenticate_sb
+#'@param session (Advanced) Alternatively supply a specific session to be checked
+#'@param ... Any additional parameters are ignored. 
+#'
+#'@description
+#'This checks the state of your Sciencebase session and, if invalid or if you've supplied
+#'a new user name, it will re-authenticate. This is primarily a helper function for 
+#'packages that reverse-depend on sbtools.
+#'
+#'@examples
+#'
+#'\dontrun{
+#'
+#'#will prompt for username/pass if needed
+#'session_check_reauth()
+#'
+#'session_check_reauth('user@@usgs.gov')
+#'
+#'#default behavior
+#'session_check_reauth(session=current_session())
+#'
+#'}
+#'
+#'
+#'@export
+session_check_reauth = function(username, password, session, ...){
```

Comment:
Hmm, could have sbtools call it. Need to break some interfaces in sbtools. 

The question is, is it better to have sbtools pushing user dialog interactions to the top, or to have it operate more fundamentally and later tools handle such user interaction?